### PR TITLE
[WX-1168] TES Log Paths

### PR DESCRIPTION
**Pull request by THWiseman**
When using a TES backend, Cromwell will now include two additional paths in the call metadata. These paths point to log files that Azure TES creates during call execution.

### CODE REVIEW

File: supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesJobCachingActorHelper.scala [hunk 1]
```diff
@@ -19,5 +19,6 @@ trait TesJobCachingActorHelper extends StandardCachingActorHelper {
   lazy val tesConfiguration: TesConfiguration = initializationData.tesConfiguration
 
   lazy val runtimeAttributes = TesRuntimeAttributes(validatedRuntimeAttributes, jobDescriptor.runtimeAttributes, tesConfiguration)
+  override protected def nonStandardMetadata: Map[String, Any] = super.nonStandardMetadata ++ tesJobPaths.azureLogPathsForMetadata
```
Comment (aednichols) [inline supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesJobCachingActorHelper.scala:22]: I love this naming and I'm glad you found the perfect fit to stick some extra stuff

File: supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesJobPaths.scala [hunk 2]
```diff
@@ -42,6 +43,13 @@ case class TesJobPaths private[tes] (override val workflowPaths: TesWorkflowPath
     case anyOtherPath: Path => anyOtherPath.pathAsString
   }
 
+  // Like above: Nothing should rely on these files existing, since only the Azure TES implementation will actually create them.
+  // Used to send the Azure TES log paths to the frontend.
+  val azureLogPathsForMetadata : Map[String, Any] = Map(
+    "tes_stdout" -> taskFullPath./("stdout.txt").pathAsString,
+    "tes_stderr" -> taskFullPath./("stderr.txt").pathAsString
```
Comment (aednichols) [inline supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesJobPaths.scala:50]: I'm wondering if the naming should reflect the purpose (frontend) but could use some context. Maybe let's sync with faces so I can get context quickly and contain the length of the semantics discussion 🙂

Semi-related, we have a bunch of metadata keys in one place [here](https://github.com/broadinstitute/cromwell/blob/7ab5582da138969126e7eb5c7267dbc66a8a78a8/services/src/main/scala/cromwell/services/metadata/CallMetadataKeys.scala#L5-L28), it's unclear whether this backend-specific one actually belongs in the main list though.

**Review by aednichols — APPROVED** [review]
LGTM 👍

Suggested a 100% optional discussion for my own understanding + naming

File: supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesJobPaths.scala (hunk 2 above)
Comment (JVThomas) [reply supportedBackends/tes/src/main/scala/cromwell/backend/impl/tes/TesJobPaths.scala:50]: Tangentially related comment: There's a method called `queryLogs` within `MetadataDatabaseAccess.scala` that currently looks for logs with specific keys (specifically "stdout" and "stderr" as two of them). The method basically queries for log-related metadata and processes that for the JSON response.

With two new keys we'd need to update these types of fetches (or any method that might use `CallMetadataKeys` to paint the metadata response) to account for these new potential keys. That said, its best to tackle these updates in another ticket.

**Review by JVThomas — APPROVED** [review]

**Pull request merged**